cmd/ls/ls_unix.go

```diff
@@ -0,0 +1,112 @@
+// +build linux
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+const (
+	_          = iota
+	KB float64 = 1 << (10 * iota)
+	MB
+	GB
+)
+
+type formatter func(os.FileInfo) (string, error)
+
+func humanizeSize(size int64) string {
+	switch sz := float64(size); {
+	case (sz >= GB):
+		return fmt.Sprintf("%.2fG", sz/GB)
+	case (sz >= MB):
+		return fmt.Sprintf("%.2fM", sz/MB)
+	case (sz >= KB):
+		return fmt.Sprintf("%.2fK", sz/KB)
+	}
+
+	return fmt.Sprintf("%d", size)
+}
+
+func formatFileName(name string) string {
+	if strings.Contains(name, " ") {
+		return "'" + name + "'"
+	}
+	return name
+}
+
+func printFileList(fileInfo os.FileInfo) (string, error) {
+	stat, ok := fileInfo.Sys().(*syscall.Stat_t)
```

Comment:
this is the only system-dependent line of this file...

@cadicallegari I think this function is the only one that you need to put inside `ls_unix.go`. Everything else should be inside `ls.go`.

Then, in the future if someone needs to port our ls to windows (for example) he/she onty needs to implement this `printFileList` function there.